File: android/src/main/java/expo/modules/deviceflags/ExpoDeviceFlagsModule.kt
package expo.modules.deviceflags

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoDeviceFlagsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoDeviceFlags")

    Constant("PI") {
      Math.PI
    }

    Function("hello") {
      "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { value: String ->
    }
  }
}

File: ios/ExpoDeviceFlagsModule.swift
import ExpoModulesCore

public class ExpoDeviceFlagsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoDeviceFlags")

    Constant("PI") {
      Double.pi
    }

    Function("hello") {
      return "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { (value: String) in
    }
  }
}
